src/Microsoft.AspNetCore.Routing/Matching/JumpTableBuilder.cs

```diff
@@ -45,12 +52,23 @@ public JumpTable Build()
                 return new ZeroEntryJumpTable(DefaultDestination, ExitDestination);
             }
 
+            if (_entries.Count == 1 && !HasNonAsciiEntry)
```

Comment:
nit: Have `if (!HasNonAsciiEntry) { return SingleEntryAsciiJumpTable } else { return SingleEntryJumpTable }` nested in count check. Likewise with linear jump table.

Reply:
meh, I think it's clearer this way.